### PR TITLE
Add ALTS to interop tests

### PR DESCRIPTION
**Pull request by jiangtaoli2016**
In order to prepare ALTS kokoro interop test, we add '--use_alts' option to test_client and test_server.

### CODE REVIEW

**Comment by jiangtaoli2016** [comment]
Oops, this PR breaks GAE interop tests due to GAE cannot use java 8 whereas ALTS uses java 8. Let me fix.

**Comment by ejona86** [comment]
This is all it took me to use Java 7 source/target (note that you can still use Java 8 libraries) such that appengine jre 7 would work. The tests would still need updating.

```patch
diff --git a/alts/build.gradle b/alts/build.gradle
index d903a4b24..6656390b9 100644
--- a/alts/build.gradle
+++ b/alts/build.gradle
@@ -1,7 +1,7 @@
 description = "gRPC: ALTS"
 
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
 
 buildscript {
     repositories {
diff --git a/alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java b/alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java
index 0ce15ac98..72a9668cc 100644
--- a/alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java
+++ b/alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java
@@ -150,7 +150,7 @@ public final class AltsChannelBuilder extends ForwardingChannelBuilder<AltsChann
 
     @Override
     public TransportCreationParamsFilter create(
-        final SocketAddress serverAddress,
+        SocketAddress serverAddress,
         final String authority,
         final String userAgent,
         final ProxyParameters proxy) {
diff --git a/alts/src/main/java/io/grpc/alts/AltsProtocolNegotiator.java b/alts/src/main/java/io/grpc/alts/AltsProtocolNegotiator.java
index 7169e8085..adfe0e1b3 100644
--- a/alts/src/main/java/io/grpc/alts/AltsProtocolNegotiator.java
+++ b/alts/src/main/java/io/grpc/alts/AltsProtocolNegotiator.java
@@ -53,7 +53,7 @@ public abstract class AltsProtocolNegotiator implements ProtocolNegotiator {
   }
 
   /** Creates a negotiator used for ALTS. */
-  public static AltsProtocolNegotiator create(final TsiHandshakerFactory handshakerFactory) {
+  public static AltsProtocolNegotiator create(TsiHandshakerFactory handshakerFactory) {
     return new AltsProtocolNegotiator() {
       @Override
       public Handler newHandler(GrpcHttp2ConnectionHandler grpcHandler) {
diff --git a/alts/src/main/java/io/grpc/alts/InternalTsiFrameHandler.java b/alts/src/main/java/io/grpc/alts/InternalTsiFrameHandler.java
index 0d8f1682a..7eefd4432 100644
--- a/alts/src/main/java/io/grpc/alts/InternalTsiFrameHandler.java
+++ b/alts/src/main/java/io/grpc/alts/InternalTsiFrameHandler.java
@@ -146,9 +146,9 @@ public final class InternalTsiFrameHandler extends ByteToMessageDecoder
   }
 
   @Override
-  public void flush(final ChannelHandlerContext ctx) throws GeneralSecurityException {
+  public void flush(ChannelHandlerContext ctx) throws GeneralSecurityException {
     checkState(protector != null, "Cannot write frames while the TSI handshake is in progress");
-    final ProtectedPromise aggregatePromise =
+    ProtectedPromise aggregatePromise =
         new ProtectedPromise(ctx.channel(), ctx.executor(), pendingUnprotectedWrites.size());
 
     List<ByteBuf> bufs = new ArrayList<>(pendingUnprotectedWrites.size());
@@ -168,11 +168,7 @@ public final class InternalTsiFrameHandler extends ByteToMessageDecoder
     }
 
     protector.protectFlush(
-        bufs, new java.util.function.Consumer<ByteBuf>() {
-          @Override public void accept(ByteBuf b) {
-            ctx.writeAndFlush(b, aggregatePromise.newPromise());
-          }
-        }, ctx.alloc());
+        bufs, b -> ctx.writeAndFlush(b, aggregatePromise.newPromise()), ctx.alloc());
 
     // We're done writing, start the flow of promise events.
     @SuppressWarnings("unused") // go/futurereturn-lsc
```

**Comment by jiangtaoli2016** [comment]
@carl-mastrangelo I just update the PR to make ALTS compatible with Java 1.7 without changing the interface. PTAL

**Comment by jiangtaoli2016** [comment]
The failed windows test seems unrelated.

File: alts/src/test/java/io/grpc/alts/AltsProtocolNegotiatorTest.java [hunk 1]
```diff
@@ -89,7 +90,8 @@
   private volatile InternalTsiHandshakeHandler.TsiHandshakeCompletionEvent tsiEvent;
   private ChannelHandler handler;
 
-  private TsiPeer mockedTsiPeer = new TsiPeer(Collections.emptyList());
+  private List<Property<?>> emptyList = Collections.emptyList();
+  private TsiPeer mockedTsiPeer = new TsiPeer(emptyList);
```
Comment (carl-mastrangelo) [inline alts/src/test/java/io/grpc/alts/AltsProtocolNegotiatorTest.java]: return this directly?

```
private TsiPeer mockedTsiPeer = new TsiPeer(Collections.<Property<?>>emptyList());
```

File: alts/src/test/java/io/grpc/alts/transportsecurity/FakeTsiHandshaker.java [hunk 2]
```diff
@@ -203,7 +205,8 @@ public boolean isInProgress() {
 
   @Override
   public TsiPeer extractPeer() {
-    return new TsiPeer(Collections.emptyList());
+    List<Property<?>> emptyList = Collections.emptyList();
+    return new TsiPeer(emptyList);
```
Comment (carl-mastrangelo) [inline alts/src/test/java/io/grpc/alts/transportsecurity/FakeTsiHandshaker.java]: same comment

File: alts/src/test/java/io/grpc/alts/AltsProtocolNegotiatorTest.java (hunk 1 above)
Comment (jiangtaoli2016) [reply alts/src/test/java/io/grpc/alts/AltsProtocolNegotiatorTest.java]: revised.

File: alts/src/test/java/io/grpc/alts/transportsecurity/FakeTsiHandshaker.java (hunk 2 above)
Comment (jiangtaoli2016) [reply alts/src/test/java/io/grpc/alts/transportsecurity/FakeTsiHandshaker.java]: revised

**Comment by jiangtaoli2016** [comment]
Code revised

**Pull request merged**